src/lib/InstantSearch.ts

```diff
@@ -184,6 +185,18 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
       );
     }
 
+    warning(
+      Boolean(insightsClient) || !hasDetectedInsightsClient(),
```

Comment:
I feel like this reads complicated, but it's because `warning` accepts a falsy argument

Reply:
I think there might be a typo in your comment. Do you mean _"`warning` accepts a boolean argument_"?

If yes, then indeed you get typescript compile error if the `condition` argument of the warning is not a strictly a boolean.